### PR TITLE
Add elevation data to cities

### LINKED ISSUE #14: Add elevation data and relative filters

### PR DESCRIPTION
**Pull request by sarslanoglu**
### Related github issue for this PR

Resolves #014

### Checklist

* [x] I have performed a self-review of my own code
* [x] I added comments, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation `README.md`
* [x] I added tests that prove my fix is effective or that my feature works
* [x] `bundle exec rake` passes locally
* [x] `rubocop` gives no error locally
* [x] My PR title includes "WIP" or is [draft PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) if work is in progress

### Type of change

* New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: lib/turkish_cities/elevation.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class Elevation
+  include DecomposerHelper
+
+  file_path = File.join(File.dirname(__FILE__), 'data/cities.yaml')
+  CITY_LIST = YAML.load_file(file_path)
+
+  def find_by_elevation(type, elevation_one, elevation_two)
+    return [] unless check_number_range(elevation_one, 4, 1923)
```
Comment (serpilacar) [inline lib/turkish_cities/elevation.rb:10]: Did you forgot to update the min-max values here? It doesn't make sense to limit it between 4-1923. I feel like we don't even need to limit the range at all, is there a specific reason? 🤔

Comment (sarslanoglu) [reply lib/turkish_cities/elevation.rb:10]: It's actually coming from data. City with lowest elevation data is Kocaeli with 4 meters and highest is Erzurum with 1923. Making a search call outside this spec won't make sense. Maybe more robust solution can be an exception with message to user

Comment (serpilacar) [reply lib/turkish_cities/elevation.rb:10]: Then how does the examples for this method on Readme really output some cities?
e.g: https://github.com/sarslanoglu/turkish_cities/pull/115/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R400

Comment (sarslanoglu) [reply lib/turkish_cities/elevation.rb:10]: Oh my bad. Readme is outdated. I will update asap

Comment (serpilacar) [reply lib/turkish_cities/elevation.rb:10]: And an exception just like you explained above in your message would be nice 💯

File: README.md [hunk 2]
```diff
@@ -374,18 +376,18 @@ Search city name with below elevation
...
+TurkishCities.find_by_elevation('below', 2213)
 # => ["Adana", "Adıyaman" ... "Kilis", "Osmaniye", "Düzce"]
-TurkishCities.find_by_elevation('woleb', 2130432)
-# => "Population type 'woleb' is unsupported"
-TurkishCities.find_by_elevation('below', 10432)
-# => []
-TurkishCities.find_by_elevation('below', 22130432)
-# => []
+TurkishCities.find_by_elevation('woleb', 213)
+# => "Elevation type 'woleb' is unsupported"
```
[the remark this answers is not in the document]
Comment (serpilacar) [reply README.md:388]: Or we can merge this for now and create a separate task to update return types on other methods as well.

**Review by serpilacar — APPROVED** [review]

File: lib/turkish_cities/population.rb [hunk 3]
```diff
@@ -28,13 +24,16 @@ def find_population(type, population_one, population_two)
 
     case type
     when 'exact'
+      check_input_range(population_one, 83_644, 15_840_901)
```
Comment (serpilacar) [inline lib/turkish_cities/population.rb:27]: It would be better to keep these numbers as a constant.

**Pull request merged**